controllers/appmesh/cloudmap_controller.go

```diff
@@ -0,0 +1,559 @@
+
+	//Reconcile pod instances with Cloudmap
+	if err := r.cloudMapInstanceReconciler.ReconcileCloudMapInstances(ctx, readyPods, notReadyPods, ignoredPods,
+		serviceSummary.ServiceID, serviceSummary.NamespaceID, vNode); err != nil {
+		log.Error(err, "Error reconciling instances with CloudMap")
+	}
+	return nil
+}
+
+func (r *cloudMapReconciler) deleteCloudMapResources(ctx context.Context, vNode *appmesh.VirtualNode) error {
+	if k8s.HasFinalizer(vNode, k8s.FinalizerAWSCloudMapResources) {
+		cloudmapConfig := vNode.Spec.ServiceDiscovery.AWSCloudMap
+		if err := r.deleteCloudMapService(ctx, vNode, cloudmapConfig); err != nil {
+			return err
+		}
+		if err := r.finalizerManager.RemoveFinalizers(ctx, vNode, k8s.FinalizerAWSCloudMapResources); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *cloudMapReconciler) getCloudMapNameSpaceFromCache(ctx context.Context,
+	key string) (cloudMapNamespaceSummary, error) {
+	existingItem, exists := r.nameSpaceIDCache.Get(key)
+	var namespaceSummary cloudMapNamespaceSummary
+	if exists {
+		//return &(existingItem.(*cloudmapNamespaceCacheItem)).value, nil
+		namespaceSummary = existingItem.(cloudMapNamespaceSummary)
+	}
+	return namespaceSummary, nil
+}
+
+func (r *cloudMapReconciler) getCloudMapNameSpaceFromAWS(ctx context.Context,
+	key string, cloudMapConfig *appmesh.AWSCloudMapServiceDiscovery) (*cloudMapNamespaceSummary, error) {
+	listNamespacesInput := &servicediscovery.ListNamespacesInput{}
+	var namespaceSummary cloudMapNamespaceSummary
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*awsservices.ListNamespacesPagesTimeout)
+	defer cancel()
+
+	err := r.cloudMapSDK.ListNamespacesPagesWithContext(ctx,
+		listNamespacesInput,
+		func(listNamespacesOutput *servicediscovery.ListNamespacesOutput, lastPage bool) bool {
+			for _, ns := range listNamespacesOutput.Namespaces {
+				if awssdk.StringValue(ns.Name) == awssdk.StringValue(&cloudMapConfig.NamespaceName) {
+					key := key
+					namespaceSummary = cloudMapNamespaceSummary{
+						NamespaceID:   awssdk.StringValue(ns.Id),
+						NamespaceType: awssdk.StringValue(ns.Type),
+					}
+
+					r.log.V(4).Info("NameSpace found ", "key: ", key, "namespaceID: ", namespaceSummary.NamespaceID)
+					r.nameSpaceIDCache.Add(key, namespaceSummary, cloudMapNamespaceCacheTTL)
+					return true
+				}
+			}
+			return false
+		},
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if namespaceSummary.NamespaceID == "" {
+		klog.Infof("No namespace found with name %s", awssdk.StringValue(&cloudMapConfig.NamespaceName))
+		return nil, nil
+	}
+	return &namespaceSummary, err
+}
+
+func (r *cloudMapReconciler) getCloudMapNameSpace(ctx context.Context,
+	cloudMapConfig *appmesh.AWSCloudMapServiceDiscovery) (cloudMapNamespaceSummary, error) {
+	key := r.namespaceCacheKey(cloudMapConfig)
+
+	if namespaceSummary, _ := r.getCloudMapNameSpaceFromCache(ctx, key); namespaceSummary.NamespaceID != "" {
```

Comment:
tbh, i'd favor to just getCloudMapNamespaceFromCache(ctx, key) to return namespaceSummary & exists.